libsolidity/Types.cpp

```diff
@@ -323,6 +323,16 @@ TypePointer IntegerConstantType::binaryOperatorResult(Token::Value _operator, Ty
 				return TypePointer();
 			value = m_value % other.m_value;
 			break;
+		case Token::EXP:
+			if (other.m_value < 0)
+				BOOST_THROW_EXCEPTION(InternalCompilerError() << errinfo_comment("exponent can't be negative"));
+			else
+			{
+				value = boost::multiprecision::powm(m_value, other.m_value, bigint(2) << 256);
```

Comment:
This is a special feature of integer literals in Solidity: https://github.com/ethereum/wiki/wiki/Solidity-Tutorial#integer-literals
According to the Solidity specification, you can do something like `unit8 x = 2**(2**(2**(2**10)))) * 0` without a type error (though the compiler will probably not be able to generate the code).